modules/core/src/v2/coins/bsv.ts

```diff
@@ -47,31 +48,13 @@ export class Bsv extends Bch {
     return common.Environments[this.bitgo.getEnv()].bsvExplorerBaseUrl + url;
   }
 
-  getAddressInfoFromExplorer(addressBase58: string): Bluebird<any> {
-    const self = this;
-    return co(function* getAddressInfoFromExplorer() {
-      // TODO BG-9989: Update this method with the correct API route and parsing once we have one
-      const addrInfo = yield request.get(self.recoveryBlockchainExplorerUrl(`/addr/${addressBase58}`)).result();
-
-      addrInfo.txCount = addrInfo.txApperances;
-      addrInfo.totalBalance = addrInfo.balanceSat;
-
-      return addrInfo;
-    }).call(this);
+  async getAddressInfoFromExplorer(addressBase58: string, apiKey?: string): Bluebird<AddressInfo> {
```

Comment:
Actually sorry, one more thing. Please remove the `async` keyword here since they can only return ES6 promises, and wrapping in `Bluebird.resolve()` doesn't actually cause a bluebird promise to be returned.

Since we don't need to use `await` here, we can just get rid of the keyword and we don't need to replace with a coroutine.

Reply:
Thanks for pointing this out!

The bluebird doc for `resolve()` says "Create a promise that is resolved with the given value. If value is already a trusted Promise, it is returned as is. If value is not a thenable, a fulfilled Promise is returned with value as its fulfillment value. If value is a thenable (Promise-like object, like those returned by jQuery's $.ajax), returns a trusted Promise that assimilates the state of the thenable." 

Why did you say `Bluebird.resolve()` doesnt actually return a promise? Or did you mean that wrapping specifically `ES6` promise in `Bluebird.resolve()` doesn't return promises?

Reply:
Right, `Bluebird.resolve()` will always return a bluebird promise. But if you return that bluebird promise from an async function, it gets converted to an ES prom, due to the magic of async/await.

Maybe this code snippet will help clarify:
```
const bb = require('bluebird');

const isBBProm = (prom) => prom._bitField !== undefined;

const bbProm = bb.resolve(1);
const esProm = Promise.resolve(2);

console.log('bbProm is bluebird:', isBBProm(bbProm));
console.log('esProm is bluebird:', isBBProm(esProm));

async function returnsBBProm() {
  return bbProm;
}

async function returnsESProm() {
  return esProm;
}

console.log('returnsBBProm() is bluebird:', isBBProm(returnsBBProm()));
console.log('returnsESProm() is bluebird:', isBBProm(returnsESProm()));
```

This outputs
```
bbProm is bluebird: true
esProm is bluebird: false
returnsBBProm() is bluebird: false
returnsESProm() is bluebird: false
```

Reply:
Thank you so much for this! Got it :)

Reply:
however I believe `bb.method(async function () { ... })` wraps it properly

Reply:
```
> (B.method(async (x) => x))(2).then(console.log)
Promise [Object] {
  _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined }
> 2
```